app/controllers/api/mixins/resource_approve_deny.rb

```diff
@@ -0,0 +1,25 @@
+module Api
+  module Mixins
+    module ResourceApproveDeny
+      def approve_resource(type, id, data)
+        reason = data["reason"]
+        api_resource(type, id, "Approving") do |resource|
+          raise BadRequestError, "Must specify a reason for approving a #{type}" if reason.blank?
+
+          resource.approve(User.current_userid, reason)
+          {}
+        end
+      end
+
+      def deny_resource(type, id, data)
+        reason = data["reason"]
+        api_resource(type, id, "Denying") do |resource|
```

Comment:
I will do this.

Do note: A few places in the code may read funny as the string is used in multiple places for failures

~ Denied Provision Request: id is required
~ Denied Provision Request: :deny is not supported (in the case where we use `supports?`

Reply:
Interesting...though I think that's unavoidable.

Reply:
Are you ok with this bad tense usage?

I had code that had a hash with the `{:delete => "Deleting", :update => "Updating"}`. We could go back to this route adding another tense. If there was no `task_id`, it could change to past tense.

I went away from this since there seemed to be a number of special cases. Right now it feels like it isn't so bad.

I could pass 2 words in for the interface but also provide some default values for the basic 4/6 actions (`type`).

Reply:
yeah let's keep it as Denying/Approving for now.  It's too messy otherwise.